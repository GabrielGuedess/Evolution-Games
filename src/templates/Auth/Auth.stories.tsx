import { Story, Meta } from '@storybook/react/types-6-0';

import { Auth, AuthProps } from './Auth';

export default {
  title: 'Templates/Auth',
  component: Auth,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta<AuthProps>;

export const Default: Story<AuthProps> = () => <Auth>Auth</Auth>;
