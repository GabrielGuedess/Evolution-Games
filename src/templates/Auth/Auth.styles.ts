import media from 'styled-media-query';

import styled, { css } from 'styled-components';

export const Wrapper = styled.main`
  ${({ theme }) => css`
    width: 100%;
    height: 100vh;
    display: flex;
    background-image: url('/img/sign-in-background-image.png');
    background-position: left;
    background-size: cover;
    background-repeat: no-repeat;
    padding: 0 ${theme.spacings.xsmall};
    position: relative;

    ${media.greaterThan('huge')`
      background-position: center;
      grid-template-columns: 1fr 1fr;
    `}
  `}
`;

export const IconWrapper = styled.a`
  ${({ theme }) => css`
    display: block;
    z-index: ${theme.layers.alwaysOnTop};

    .arrow {
      position: absolute;
      color: ${theme.colors.whiteText};
      top: calc(100% / 20);
      left: 0.4rem;
    }

    ${media.greaterThan('medium')`
      display: none;
    `}
  `}
`;

export const FirstColumn = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  overflow-y: auto;

  ::-webkit-scrollbar {
    display: none;
  }
`;

export const SecondColumn = styled.div`
  ${({ theme }) => css`
    width: 100%;
    height: 100%;
    display: none;
    align-items: flex-end;
    justify-content: center;
    padding-bottom: ${theme.spacings.huge};

    ${media.greaterThan('huge')`
      display: flex;
    `}
  `}
`;

export const GoBack = styled.a`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xxmedium};
    font-style: italic;
    font-weight: 300;
    line-height: ${theme.spacings.medium};
    color: ${theme.colors.whiteText};
    text-decoration: underline;
  `}
`;
