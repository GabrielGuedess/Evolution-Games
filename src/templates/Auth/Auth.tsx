import Link from 'next/link';

import { CaretLeft } from 'phosphor-react';

import * as S from './Auth.styles';

export type AuthProps = {
  children: React.ReactNode;
};

export const Auth = ({ children }: AuthProps) => (
  <S.Wrapper>
    <Link href="/" passHref>
      <S.IconWrapper>
        <CaretLeft size={32} className="arrow" />
      </S.IconWrapper>
    </Link>

    <S.FirstColumn>{children}</S.FirstColumn>

    <S.SecondColumn>
      <Link href="/" passHref>
        <S.GoBack>Voltar para a Home</S.GoBack>
      </Link>
    </S.SecondColumn>
  </S.Wrapper>
);
