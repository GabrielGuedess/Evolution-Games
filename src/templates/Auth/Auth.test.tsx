import { renderWithProviders } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import { Auth } from './Auth';

describe('<Auth />', () => {
  it('should render the children', () => {
    // Arrange
    const { container } = renderWithProviders(<Auth>Auth</Auth>);

    // Assert
    expect(screen.getByText(/Auth/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
