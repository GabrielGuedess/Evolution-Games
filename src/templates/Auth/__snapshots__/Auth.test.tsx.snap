// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Auth /> should render the children 1'] = `
.c0 {
  width: 100%;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-image: url('/img/sign-in-background-image.png');
  background-position: left;
  background-size: cover;
  background-repeat: no-repeat;
  padding: 0 1.6rem;
  position: relative;
}

.c1 {
  display: block;
  z-index: 50;
}

.c1 .arrow {
  position: absolute;
  color: #F4F4F5;
  top: calc(100% / 20);
  left: 0.4rem;
}

.c2 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow-y: auto;
}

.c2::-webkit-scrollbar {
  display: none;
}

.c3 {
  width: 100%;
  height: 100%;
  display: none;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-bottom: 8.0rem;
}

.c4 {
  font-size: 2.0rem;
  font-style: italic;
  font-weight: 300;
  line-height: 3.2rem;
  color: #F4F4F5;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media (min-width:1440px) {
  .c0 {
    background-position: center;
    grid-template-columns: 1fr 1fr;
  }
}

@media (min-width:768px) {
  .c1 {
    display: none;
  }
}

@media (min-width:1440px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<main
  class="c0"
>
  <a
    class="c1"
    href="/"
  >
    <svg
      class="arrow"
      fill="currentColor"
      height="32"
      viewBox="0 0 256 256"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="256"
        width="256"
      />
      <polyline
        fill="none"
        points="160 208 80 128 160 48"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
      />
    </svg>
  </a>
  <div
    class="c2"
  >
    Auth
  </div>
  <div
    class="c3"
  >
    <a
      class="c4"
      href="/"
    >
      Voltar para a Home
    </a>
  </div>
</main>
`;
